_assets/javascripts/components/bitmovin.js

```diff
@@ -0,0 +1,248 @@
+    this.events
+     .forEach(e => {
+        const now = moment.tz(this.timezoneStr);
+        const timeTilEventStart = moment.tz(e.start, this.timezoneStr) - now;
+        const videoEndTime = moment.tz(e.start, this.timezoneStr) + this.videoDuration;
+        const timeTilVideoEnd = videoEndTime - moment.tz(this.timezoneStr);
+        if (moment.tz(e.start, this.timezoneStr) > moment.tz(this.timezoneStr)) {
+          setTimeout(() => {
+            this.restartVideo();
+          }, timeTilEventStart);
+        }  
+
+        setTimeout(() => {
+          this.showStandbyMessaging();
+        }, timeTilVideoEnd);
+      });
+  }
+
+  getHideUI() {
+    return this.isCard;
+  }
+
+  getHidePlaybackSpeed() {
+    return this.isStream;
+  }
+
+  getAutoPlay() {
+    if (this.getStartTime() > 0 && !this.currentHasEnded()) return true;
+    let urlParams = new URLSearchParams(window.location.search);
+    let autoPlay = urlParams.has('autoplay')
+      ? Boolean(urlParams.get('autoplay'))
+      : false;
+    return autoPlay;
+  }
+
+  getIsMuted() {
+    let urlParams = new URLSearchParams(window.location.search);
+    let sound = urlParams.has('sound') ? parseInt(urlParams.get('sound')) : 0;
+    if (sound == 11) return false;
+    return true;
+  }
+
+  getStartTime() {
+    let startTime = 0;
+    if (this.countdown.currentEvent && !this.currentHasEnded()) {
+      startTime = this.calculateStreamElapsed();
+    } else {
+      let urlParams = new URLSearchParams(window.location.search);
+      let min = urlParams.has('min') ? parseInt(urlParams.get('min')) : 0;
+      let sec = urlParams.has('sec') ? parseInt(urlParams.get('sec')) : 0;
+      if (min || sec) {
+        startTime = min * 60 + sec;
+      }
+    }
+
+    return startTime;
+  }
+
+  currentHasEnded() {
+    const  now = moment.tz(this.timezoneStr);
+    return now - this.currentVideoEndTime > 0 ? true : false;
+  }
+
+  onPlayerStart() {
+    if (typeof analytics !== 'undefined') {
+      analytics.track('VideoStarted', {
+        Title: this.bitmovinPlayer.getSource().title,
+        VideoId: this.bitmovinPlayer.getSource().hls,
+        Source: 'CrossroadsNet',
+        VideoTotalDuration: this.bitmovinPlayer.getDuration()
+      });
+    }
+  }
+
+  onPlayerEnd(reason) {
+    if (typeof analytics !== 'undefined') {
+      analytics.track('VideoEnded', {
+        Title: this.bitmovinPlayer.getSource().title,
+        VideoId: this.bitmovinPlayer.getSource().hls,
+        Source: 'CrossroadsNet',
+        VideoTotalDuration: this.bitmovinPlayer.getDuration(),
+        CurrentTime: this.bitmovinPlayer.getCurrentTime(),
+        ReasonForEnding: reason
+      });
+    }
+  }
+
+  getQualityLabels(data) {
+    let resolution = '';
+    let kbps = Math.round(data.bitrate / 1000);
+
+    // convert to megabits if applicable
+    let bitrate =
+      kbps > 1000 ? `${(kbps / 1000).toFixed(1)} mbps` : `${kbps} kbps`;
+
+    if (data.height <= 240) {
+      resolution = '240p';
```

Comment:
should these be labeled `SD` as well? the standard YouTube uses is anything below 720p just doesn't get a definition label. they don't even use SD.

![YouTube Options](https://user-images.githubusercontent.com/26073074/58658943-bfe8a400-82ef-11e9-8c22-259c3121fba8.png)